eppo_core/src/events/event_delivery.rs

```diff
@@ -0,0 +1,60 @@
+use crate::events::event::Event;
+use crate::Error;
+use log::info;
+use reqwest::{Method, StatusCode};
+use serde::{Deserialize, Serialize};
+
+pub struct EventDelivery {
+    sdk_key: String,
+    ingestion_url: String,
+    client: reqwest::Client,
+}
+
+#[derive(serde::Deserialize)]
+struct EventDeliveryResponse {
+    failed_events: Vec<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct IngestionRequestBody {
+    eppo_events: Vec<Event>,
+}
+
+/// Responsible for delivering event batches to the Eppo ingestion service.
+impl EventDelivery {
+    pub fn new(sdk_key: String, ingestion_url: String) -> Self {
+        let client = reqwest::Client::new();
+        EventDelivery {
+            sdk_key,
+            ingestion_url,
+            client,
+        }
+    }
+
+    // Delivers the provided event batch and returns a Vec with the events that failed to be delivered.
+    pub async fn deliver(self, events: &[Event]) -> Result<Vec<String>, Error> {
+        let ingestion_url = self.ingestion_url;
+        let sdk_key = self.sdk_key;
+        info!("Delivering {} events to {}", events.len(), ingestion_url);
+        let body = IngestionRequestBody { eppo_events: events.to_vec() };
+        let serialized_body = serde_json::to_string(&body).expect("Failed to serialize body");
+        let response = self.client.request(Method::POST, ingestion_url)
+            .header("Content-Type", "application/json")
+            .header("X-Eppo-Token", sdk_key)
+            .body(serialized_body)
+            .send()
+            .await?;
```

Comment:
nit: reqwest has `.post()` and `.json()` shortcuts:
```suggestion
        let response = self.client.post(ingestion_url)
            .header("X-Eppo-Token", sdk_key)
            .json(&body)
            .send()
            .await?;
```